fms/utils/serialization.py

```diff
@@ -136,12 +188,270 @@ def load_state_dict(
 
         checkpoints = [checkpoints[rank]]
 
-    checkpoint_sds = [
-        torch.load(ckpt_path, map_location="cpu") for ckpt_path in checkpoints
-    ]
+    if file_format == "st":
+        from safetensors import safe_open  # type: ignore
+
+        # For safetensors, delay loading the weights until information about the model
+        # is available for sharding
+        checkpoint_sds = []
+        for ckp in checkpoints:
+            with safe_open(
+                ckp, framework="pt", device=str(initial_device)
```

Comment:
in the else, we are doing a `device="cpu"`. I noticed in a test, that initial_device ended up being "cpu:0" which failed. Might want this to follow same as the below torch load.

Reply:
`initial_device` being `cpu:0` is a bigger issue that we should solve, as it should never be that. Can you share what test created such a device?

Reply:
found how it can happen, fixed in the latest iteration of code